scylla/src/network/connection.rs

```diff
@@ -322,17 +323,44 @@ impl ConnectionConfig {
             authenticator: self.authenticator.clone(),
             address_translator: self.address_translator.clone(),
             write_coalescing_delay: self.write_coalescing_delay.clone(),
+            pending_request_channel_size: self.pending_request_channel_size,
+
             keepalive_interval: self.keepalive_interval,
             keepalive_timeout: self.keepalive_timeout,
             tablet_sender: self.tablet_sender.clone(),
             identity: self.identity.clone(),
         }
     }
+
+    /// Set the size of the pending request channel for each connection.
+    ///
+    /// # Arguments
+    ///
+    /// * `size` - The maximum number of pending requests per connection.
+    ///   
+    /// # Notes
+    /// 
+    /// - This is different from cpp-driver's implementation, which uses a per-RequestProcessor queue.
+    /// - The default is 2048, a balanced value between performance and memory usage.
+    /// - Adjust based on your specific workload and system resources.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// let session = SessionBuilder::new()
+    ///     .connection_config(
+    ///         ConnectionConfig::new()
+    ///             .with_pending_request_channel_size(4096)
+    ///     )
+    ///     .build()
+    ///     .await?;
+    /// ```
```

Comment:
I think it is worth describing what a "pending request" is. I think users may think that it is a request that was sent, but we did not yet receive response to (while in reality it is a request that has not yet been sent).